gaia-rs/src/message.rs

```diff
@@ -15,10 +18,7 @@ pub enum Message {
 }
 
 impl ValueRenderer for Message {
-    fn format<F: Fn(&Denom) -> Option<Metadata>>(
-        &self,
-        get_metadata: &F,
-    ) -> Result<Vec<Screen>, RenderError> {
+    fn format<MG: MetadataGetter>(&self, get_metadata: &MG) -> Result<Vec<Screen>, RenderError> {
```

Comment:
It's because the `get_metadata` function returns a `Result` now, which needs an error type. I found it cleaner if this was done as a trait rather than a Fn.